packages/brick-kit/src/internal/images.ts

```diff
@@ -1,18 +1,36 @@
+import { getBasePath } from "./getBasePath";
 export interface ImagesFactory {
   get(name: string): string;
 }
 
 export function imagesFactory(
   appId: string,
-  isBuildPush?: boolean
+  isBuildPush?: boolean,
+  version?: string
 ): ImagesFactory {
   return {
     get(name) {
+      const getSuffix = (): string => {
+        let suffix = window.APP_ROOT ? `${window.APP_ROOT}-/` : "";
+        if (!suffix.startsWith("/")) {
+          suffix = getBasePath() + suffix;
+        }
+        if (window.APP_ID && window.APP_ID !== appId) {
+          return suffix.replace(
+            new RegExp(`/(${window.APP_ID}|\\d+.\\d+.\\d+)/`, "g"),
+            (_, p1) => {
+              if (p1 === window.APP_ID) {
+                return `/${appId}/`;
+              }
+              return `/${version}/`;
+            }
+          );
+        }
+        return suffix;
+      };
       return isBuildPush
-        ? `api/gateway/object_store.object_store.GetObject/api/v1/objectStore/bucket/next-builder/object/${name}`
-        : `${
-            window.APP_ROOT ? `${window.APP_ROOT}-/` : ""
-          }micro-apps/${appId}/images/${name}`;
+        ? `${getSuffix()}api/gateway/object_store.object_store.GetObject/api/v1/objectStore/bucket/next-builder/object/${name}`
```

Comment:
api gateway 不需要 APP_ROOT 这些前缀...

```suggestion
        ? `${getBasePath()}api/gateway/object_store.object_store.GetObject/api/v1/objectStore/bucket/next-builder/object/${name}`
```